src/goes/goes_converter.py

```diff
@@ -174,6 +200,115 @@ def _create_metadata_scan_angle_variables(self):
+        self._output_dataset_bt['/MetaData/sensor_zenith_angle'][:] = sensor_zenith_angle_data_array
+        self._output_dataset_bt['/MetaData/sensor_zenith_angle'].setncattr('units', 'degrees')
+
+    def _create_metadata_sensor_azimuth_angle_variables(self):
+        """
+        Creates the /MetaData/sensor_azimuth_angle variable in the reflectance factor and brightness temperature netCDF4
+        Datasets.
+        """
+        sensor_azimuth_angle_data_array = self._latlon_dataset['MetaData'].variables['sensor_azimuth_angle'][:].real
+        self._output_dataset_rf.createVariable('/MetaData/sensor_azimuth_angle', 'f4', 'nlocs', fill_value=-999)
+        self._output_dataset_rf['/MetaData/sensor_azimuth_angle'][:] = sensor_azimuth_angle_data_array
+        self._output_dataset_rf['/MetaData/sensor_azimuth_angle'].setncattr('units', 'degrees')
+        self._output_dataset_bt.createVariable('/MetaData/sensor_azimuth_angle', 'f4', 'nlocs', fill_value=-999)
+        self._output_dataset_bt['/MetaData/sensor_azimuth_angle'][:] = sensor_azimuth_angle_data_array
+        self._output_dataset_bt['/MetaData/sensor_azimuth_angle'].setncattr('units', 'degrees')
+
+    def _create_metadata_sensor_view_angle_variables(self):
+        """
+        Creates the /MetaData/sensor_view_angle variable in the reflectance factor and brightness temperature netCDF4
+        Datasets.
+        """
+        sensor_view_angle_data_array = self._latlon_dataset['MetaData'].variables['sensor_view_angle'][:].real
+        self._output_dataset_rf.createVariable('/MetaData/sensor_view_angle', 'f4', 'nlocs', fill_value=-999)
+        self._output_dataset_rf['/MetaData/sensor_view_angle'][:] = sensor_view_angle_data_array
+        self._output_dataset_rf['/MetaData/sensor_view_angle'].setncattr('units', 'degrees')
+        self._output_dataset_bt.createVariable('/MetaData/sensor_view_angle', 'f4', 'nlocs', fill_value=-999)
+        self._output_dataset_bt['/MetaData/sensor_view_angle'][:] = sensor_view_angle_data_array
+        self._output_dataset_bt['/MetaData/sensor_view_angle'].setncattr('units', 'degrees')
+
+    def _create_metadata_solar_zenith_angle_variables(self):
+        """
+        Creates the /MetaData/solar_zenith_angle variable in the reflectance factor and brightness temperature netCDF4
+        Datasets.
+        """
+        dataset = Dataset(self._input_file_path_template, 'r')
+        dataset.set_auto_scale(True)
+        dataset_latlon = Dataset(self._latlon_file_path, 'r')
+        latitude = ma.getdata(dataset_latlon['/MetaData/latitude'][:].real)
+        start_date = self._start_date
+        t = start_date.hour + (start_date.minute / 60.0) + (start_date.second / 3600.0)
+        h = -1.0 * ((t - 12.0) / 12.0)
+        j = self._day_of_year
+        declin = -23.45 * np.cos(((2.0 * np.pi * float(j)) / 365.0) + ((20.0 * np.pi) / 365.0))
```

Comment:
Can you put a comment in this section of the code to identify the source of these relationships (if you know the source)?  This will help with future verification / understanding.

This comment applies to the azimuth section as well.

Reply:
The sources for all formulations used in each python class are listed in the header documentation of each class.